src/ucs/stats/stats.c

```diff
@@ -112,6 +117,59 @@ static void ucs_stats_clean_node(ucs_stats_node_t *node) {
     ucs_list_del(&node->type_list);
 }
 
+static ucs_stats_class_t *ucs_stats_dup_class(ucs_stats_class_t *cls)
+{
+    ucs_stats_class_t *dup;
+
+    dup = ucs_malloc(sizeof(*cls) + sizeof(*cls->counter_names) * cls->num_counters,
+                     "ucs_stats_class_t");
+    ucs_assertv_always(dup != NULL, "failed to allocate statistics class");
+
+    dup->name = ucs_strdup(cls->name, "ucs_stats_class_t name");
+    ucs_assertv_always(dup->name != NULL, "failed to copy statistics class name");
+
+    for (dup->num_counters = 0; dup->num_counters < cls->num_counters; dup->num_counters++) {
+        dup->counter_names[dup->num_counters] = ucs_strdup(cls->counter_names[dup->num_counters],
+                                                           "ucs_stats_class_t counter");
+        ucs_assertv_always(dup->counter_names[dup->num_counters] != NULL,
+                           "failed to copy statistics counter name");
+    }
+
+    return dup;
+}
+
+static void ucs_stats_free_class(ucs_stats_class_t *cls)
+{
+    unsigned i;
+
+    for (i = 0; i < cls->num_counters; i++) {
+        ucs_free((void*)cls->counter_names[i]);
+    }
+
+    ucs_free((void*)cls->name);
+
+    free(cls);
+}
+
+static ucs_stats_class_t *ucs_stats_get_class(ucs_stats_class_t *cls)
+{
+    ucs_stats_class_t *dup;
+    khiter_t iter;
+    int r;
+
+    iter = kh_get(ucs_stats_cls, &ucs_stats_context.cls, (uint64_t)cls);
```

Comment:
minor: you could just put and if ret_code is zero - key already added, otherwise dup and set val